File: backend/src/main/java/com/kiwi/features/tips/data/TipMapper.java
package com.kiwi.features.tips.data;

public class TipMapper {
    public static TipDomain toDomain(TipPersistence persistence) {
        return new TipDomain(persistence.getTitle(), persistence.getText(), persistence.getReadMoreURL());
    }
    
    public static TipDTO toDTO(TipDomain domain) {
        TipDTO dto = new TipDTO();
        
        dto.setTitle(domain.getTitle());
        dto.setText(domain.getText());
        dto.setReadMoreURL(domain.getReadMoreURL());
        
        return dto;
    }
    
    public static TipPersistence toPersistence(TipDomain domain) {
        TipPersistence persistence = new TipPersistence();

        persistence.setTitle(domain.getTitle());
        persistence.setText(domain.getText());
        persistence.setReadMoreURL(domain.getReadMoreURL());

        return persistence;
    }
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/tips/data/TipMapper.kt
package com.bellako.kiwi.features.tips.data

object TipMapper {
    fun toDomain(dto: TipDTO): TipDomain = TipDomain(id = dto.id, title = dto.title, text = dto.text, readMoreURL = dto.readMoreURL)

    fun toDTO(domain: TipDomain): TipDTO =
        TipDTO(id = domain.id, title = domain.title, text = domain.text, readMoreURL = domain.readMoreURL)
}
